esp/esp/program/modules/handlers/programprintables.py

```diff
@@ -398,6 +398,22 @@ def cmp_title(one, other):
 
         return self.classesbyFOO(request, tl, one, two, module, extra, prog, cmp_title)
 
+    @aux_call
+    @needs_admin
+    def classesbyteacher(self, request, tl, one, two, module, extra, prog):
+        def cmp_teacher(one, other):
+            cmp0 = cmp(one.get_teachers()[0].last_name.lstrip(), other.get_teachers()[0].last_name.lstrip())
```

Comment:
Huh, so this is interesting -- do you want to sort the teacher names?  Or do you want to change things to include an entry for each class under each teacher name?  Right now this seems like it will sort by the name of an arbitrary teacher, which is not great, and even sorting by the name of the alphabetically-first teacher is not ideal.

Reply:
Hmm...This is certainly the *easiest* way to do it, although just taking the alphabetically-first teacher would be just as easy. However, I agree it probably makes the most sense to have an entry for each teacher? But that won't be doable with `classesbyFOO` as it currently exists. I guess all we would need to do is add a little bit of code in `classesbyFOO` to duplicate classes for each teacher (and assign that teacher as an attribute).

Reply:
Yeah, it's true.  Not sure how much it matters to admins, it just seems like it would be very confusing to me to have a list of classes by teacher that doesn't list my class under my name because I have an alphabetically-earlier coteacher.